ci/Jenkinsfile-pyp

```diff
@@ -0,0 +1,205 @@
+                            curl -XPOST -H "Content-Type: application/json" -H "Authorization: token ${GITHUB_TOKEN}" --data "${request_data}" https://api.github.com/repos/inmanta/inmanta-sphinx/pulls
+
+                            # Build python package from next branch
+                            git checkout next
+                        '''
+                    }
+                }
+            }
+        }
+
+        stage("Setup next branch for patch release") {
+            when {
+                allOf {
+                    expression { "${run_job}" == "true" }
+                    expression { "${release_type}" == "stable" }
+                    expression { "${version_increment}" == "patch" }
+                }
+            }
+            steps {
+                dir('inmanta-sphinx') {
+                    withCredentials([string(credentialsId: 'fff7ef7e-cb20-4fb2-a93b-c5139463c6bf', variable: 'GITHUB_TOKEN')]) {
+                        sh '''
+                            git checkout next
+
+                            # Cherry-picking the patch commits on the next branch and setting
+                            # the changelog correctly is done manually before this job runs.
+
+                            export GIT_ASKPASS="${WORKSPACE}/ASK_PASS"
+                            ${WORKSPACE}/env/bin/python3 -m pip install -U pip bumpversion
+
+                            # Bump version
+                            ${WORKSPACE}/env/bin/bumpversion --commit ${version_increment}
+                            git push origin next
+                        '''
+                    }
+                }
+            }
+        }
+
+        stage("Add tag") {
+            when {
+                allOf {
+                    expression { "${run_job}" == "true" }
+                    expression { "${release_type}" == "stable" }
+                }
+            }
+            steps{
+                dir('inmanta-sphinx') {
+                    withCredentials([string(credentialsId: 'fff7ef7e-cb20-4fb2-a93b-c5139463c6bf', variable: 'GITHUB_TOKEN')]) {
+                        sh '''
+                            export GIT_ASKPASS="${WORKSPACE}/ASK_PASS"
+                            current_version=$(sed -n -E 's/current_version[[:space:]]*=[[:space:]]*([\\digit.]*)/\\1/p' .bumpversion.cfg)
+                            tag="v${current_version}"
+                            git tag -a -m "Release: ${current_version}" ${tag}
+                            git push origin ${tag}
+                        '''
+                    }
+                }
+            }
+        }
+
+        stage("Build dist") {
+            when {
+                expression { "${run_job}" == "true" }
+                anyOf {
+                    expression { "${release_type}" == "stable" }
+                    expression { "${release_type}" == "dev" }
+                }
```

Comment:
This condition can be removed. It's an exhaustive list of all possible values for release_type.

Reply:
I am aware. I added it as a guard against future expansion of `release_type` so we can be sure that the `else` branch is `dev`. Should I remove it?

Reply:
Ok. You can keep it like this.